workflows/rnaseq/downstream/rnaseq.Rmd

```diff
@@ -395,32 +403,48 @@ res.list <- list()
 #   Using the example data, this compares treatment group to control group.
 #   Change to reflect your experiment.
 # IMPORTANT: Control must always be last
+res.all <- results(
+    dds,
+    contrast=c('group', 'treatment', 'control'),
+    alpha=alpha,
+    lfcThreshold=0)
+
 res.list[['all']] <- list(
-    res=lfcShrink(dds, contrast=c('group', 'treatment', 'control'), type='normal'),
+    res=lfcShrink(dds,
+                  contrast=c('group', 'treatment', 'control'),
+                  alpha=metadata(res.all)[['alpha']],
+                  res=res.all,
+                  type='normal'
+                  ),
     dds=dds,
-    label='Using a log2FoldChange threshold of 0'
+    label='Using a log2FoldChange threshold of 0',
+    alpha=metadata(res.all)[['alpha']],
+    lfc.thresh=metadata(res.all)[['lfcThreshold']]
```

Comment:
So we can get the alpha and lfcThreshold from the results object using `metadata()`. . . I think rather than have it here, it would be cleaner to instead pull out this info from inside any helper functions that operate on res.list or on a results object. I mean, that complexity to extract alpha and lfcThreshold does have to exist somewhere; I'd rather not have it here where the user needs to type it in each time.

Reply:
I agree with Ryan here. I like the idea of using the `metadata()` function to unify the lfc and fdr thresholds in the helper functions. That will definitely clean up the code and we won't need to set it explicitly here.

Reply:
`lfcShrink` does not carry over the res lfcThreshold. Would it make more sense to do the `results()` without threshold and apply an lfcThreshold to `lfcShrink`? Or to apply the lfcThreshold to both `results()` and `lfcShrink`?

Reply:
Oh no!

It might take writing it out and seeing it both ways to figure out what the best long-term solution will be. I think the major guiding principle is simplicity on the rnaseq.Rmd side (to avoid bugs). If it doesn't become clear after trying it different ways, I think having a separate PR ("dueling pull requests") would be a good way for us to jointly evaluate the options.

Reply:
The lfcThreshold in `lfcShrink()` is for s-values. Michael Love recommends setting the threshold in `results()` to work with an FDR bounded set ([here](https://support.bioconductor.org/p/110307/)), but that means we can't use the metadata from the lfcShunk res.list to figure out the lfcThreshold.